_posts/2024-03-22-dependency-management.md

```diff
@@ -0,0 +1,281 @@
+that both inlining and rewriting come with serious drawbacks. The
+most fundamental issue is probably related to code maintenance. Every significant piece of software needs some level of
+maintenance. Ranging from fixing bugs to dealing with changing (build) environments or requirements, code will
+need to be modified to stay functional and relevant. When inlining/rewriting dependencies, this also copies over the
+maintenance burden.
+
+For DuckDB, this historically meant that for each dependency, a very careful consideration was made balancing the
+increased maintenance burden against the necessity of dependency. Including a dependency meant the responsibility of
+maintaining it, so this decision was never taken lightly. This works well in many cases and has the advantage of forcing
+developers to critically think about including a dependency and not mindlessly bolt on library after library. However, 
+for some dependencies, this just doesn’t work. Large cloud provider SDKs for example. They tend to be pretty
+massive, very frequently updated, and packed with arguably essential functionality for an increasingly mature analytical
+database. This leaves an awkward choice: either not provide these essential features, or break the no-dependency rule.
+
+## DuckDB Extensions
+This is where extensions come in. Extensions provide an elegant solution to the dilemma of dependencies, by allowing
+fine-grained breakage of the no-dependency rule. Moving dependencies out of DuckDB’s core, into extensions, the core 
+codebase can remain, and does remain, dependency-free.
+This means that DuckDB’s “Practical embeddability and portability” remains unthreatened. On the other hand, DuckDB can
+still provide features that inevitably require depending on some 3rd party library. Furthermore, by moving dependencies
+to extensions, each extension can have different levels of exposure to instability from dependencies. For example, some
+extensions may choose to only depend on highly mature, stable libraries with good portability, whereas others may choose
+to include more experimental dependencies with limited portability. This choice is then forwarded to the user by
+allowing them to choose which extension to use.
+
+At DuckDB, this realization of the importance of extensions and its relation to the no-dependency rule came
+[very early](https://github.com/duckdb/duckdb/pull/594), and consequently extensibility has been ingrained into DuckDB’s
+design from its early days. Today, many parts of DuckDB can be extended. For example, you can add functions (table,
+scalar, copy, aggregation), filesystems, parsers, optimizer rules, and much more. Many new features that are added to
+DuckDB are added in extensions and are grouped by either functionality, or by set of dependencies. Some examples of
+extensions are the [JSON](/docs/extensions/json.html) extension for JSON support or the
```

Comment:
I would not cover JSON here, since this is not really an extension given it's basically built-in in all platforms.

Maybe sqlite_scanner is more representative?